src/test/java/book_store/controller/BookControllerTest.java

```diff
@@ -0,0 +1,233 @@
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto actualBook = objectMapper.readValue(result.getResponse().getContentAsByteArray(), BookDto.class);
+
+        Assertions.assertEquals(expectedBook.getId(), actualBook.getId());
+        Assertions.assertEquals(expectedBook.getTitle(), actualBook.getTitle());
+        Assertions.assertEquals(expectedBook.getAuthor(), actualBook.getAuthor());
+        Assertions.assertEquals(expectedBook.getPrice(), actualBook.getPrice());
+        Assertions.assertEquals(expectedBook.getIsbn(), actualBook.getIsbn());
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Sql(
+            scripts = "classpath:database/books/delete-John_Doe-book.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Create a new book")
+    void createBook_CreatesNewBook_ExpectedSuccess() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto()
+                .setAuthor("John Doe")
+                .setTitle("Book Title")
+                .setDescription("Book Description")
+                .setIsbn("9460306332081")
+                .setPrice(BigDecimal.valueOf(10.99));
+
+        BookDto expected = new BookDto()
+                .setAuthor(requestDto.getAuthor())
+                .setTitle(requestDto.getTitle())
+                .setDescription(requestDto.getDescription())
+                .setIsbn(requestDto.getIsbn())
+                .setPrice(requestDto.getPrice());
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                post("/books")
+                    .content(jsonRequest)
+                    .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        Assertions.assertEquals(expected.getAuthor(), actual.getAuthor());
+        EqualsBuilder.reflectionEquals(expected,actual,"id");
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Delete a book")
+    void delete_DeletesBookById_ExpectedSuccess() throws Exception {
+        Book book = new Book();
+                book.setId(4L);
+                book.setAuthor("John Doe");
+                book.setTitle("Book Title");
+                book.setDescription("Book Description");
+                book.setIsbn("9460306332081");
+                book.setPrice(BigDecimal.valueOf(10.99));
+        mockMvc.perform(
+                        delete("/books/{id}", 4L)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isNoContent());
+    }
```

Comment:
Why you created book? You don't use it further.
Also would be nice to recheck DB state, if book was really deleted.
If your controller has code "return new ResponseEntity(NO_CONTENT);" this test will pass